CMakeLists.txt

```diff
@@ -297,7 +297,8 @@ target_compile_options(
             # Never treat deprecation warnings as errors.
             -Wno-error=deprecated
             -Wno-error=deprecated-declarations>
-            $<$<CXX_COMPILER_ID:GNU>:-Wno-redundant-move>)
+            $<$<CXX_COMPILER_ID:GNU>:-Wno-redundant-move>
+            $<$<CXX_COMPILER_ID:GNU>:-Wno-error=bool-compare>)
```

Comment:
I believe this would be better solved by not allowing bool as tParam for the `integral_printer`. The dedicated `bool_printer` should be used anyway.

Reply:
That's not quite the problem. The problem is that `integral_printer<T>::print` method is called with a `bool` for its argument. The class template type for the instantiation of `integral_printer` is non-bool.